core/store/src/trie/update.rs

```diff
@@ -136,11 +136,26 @@ impl TrieUpdate {
                     .clone();
                 state_changes.push(changes_with_trie_key);
                 (k, data)
-            }),
-        )?;
+            }))?;
         Ok((trie_changes, state_changes))
     }
 
+    pub fn finalize_genesis(self) -> Result<TrieChanges, StorageError> {
+        assert!(self.prospective.is_empty(), "Finalize cannot be called with uncommitted changes.");
+        let TrieUpdate { trie, committed, .. } = self;
+        let trie_changes =
+            trie.update(committed.into_iter().map(|(k, changes_with_trie_key)| {
+                let data = changes_with_trie_key
+                    .changes
+                    .into_iter()
+                    .last()
+                    .expect("Committed entry should have at least one change")
+                    .data;
+                (k, data)
+            }))?;
+        Ok(trie_changes)
+    }
```

Comment:
Why we need this? It seems a duplicate of `finalize` and also is unused.

Reply:
Good catch.  Looks like it was bad merge.